src/cli/commands/test/iac-local-execution/parsers/terraform-plan-parser.ts

```diff
@@ -125,8 +127,21 @@ export function tryParsingTerraformPlan(
       },
     ];
   } catch (err) {
-    throw new Error(
-      'Failed to extract resources from Terraform plan JSON file.',
-    );
+    throw new MissingRequiredFieldsInTerraformPlanError();
+  }
+}
+
+class FailedToParseTerraformPlanJsonError extends CustomError {
+  constructor(message?: string) {
+    super(message || 'Failed to parse Terraform plan JSON file');
+    this.code = IaCErrorCodes.FailedToParseTerraformPlanJsonError;
+    this.userMessage = '';
+  }
+}
+class MissingRequiredFieldsInTerraformPlanError extends CustomError {
+  constructor(message?: string) {
+    super(message || 'Failed to parse Terraform plan JSON file');
+    this.code = IaCErrorCodes.MissingRequiredFieldsInTerraformPlanError;
+    this.userMessage = '';
```

Comment:
Needs the fields added.

```suggestion
    this.userMessage = 'We failed to scan the provided Terraform plan file, it was expect to contain fields x, y & z, please contact support@snyk.io, if possible with a redacted version of the file';
```